cpp/src/arrow/compute/exec/asof_join_benchmark.cc

```diff
@@ -77,14 +88,18 @@ static void TableJoinOverhead(benchmark::State& state,
     state.PauseTiming();
     ASSERT_OK_AND_ASSIGN(std::shared_ptr<arrow::compute::ExecPlan> plan,
                          ExecPlan::Make(&ctx));
-    std::vector<ExecNode*> input_nodes = {*arrow::compute::MakeExecNode(
-        "table_source", plan.get(), {},
-        arrow::compute::TableSourceNodeOptions(left_table_stats.table, batch_size))};
+    std::vector<std::shared_ptr<arrow::internal::ThreadPool>> right_thread_pools;
+    std::shared_ptr<arrow::internal::ThreadPool> left_thread_pool =
```

Comment:
Let's add a comment (can be the same as previous) explaining why this dedicated thread pool is needed.